src/main/java/org/simulator/math/odes/RosenbrockSolver.java

```diff
@@ -440,8 +442,7 @@ public int getNumEquations() {
         hasDerivatives = false;
       }
     }
-    double timeEnd = BigDecimal.valueOf(time).add(BigDecimal.valueOf(currentStepSize))
-        .doubleValue();
+    double timeEnd = time + currentStepSize;
```

Comment:
According to the IEEE754 spec, floating point addition must already be "as precise as possible", i.e. return the closest floating point number to the real (in the sense of real numbers) result. So I don't think there is any benefit in translation to `BigDecimal` and back.